File: rulngin-ui/src/features/booleanStatement/entity/BooleanPropertyStatement.ts
import {PropertyStatement} from "../../statement/entity/Statement.ts";

export type BooleanPropertyOperator = 'EQUALS' | 'NOT_EQUALS';

export type BooleanPropertyStatement = PropertyStatement & {
  statementType: 'BooleanPropertyStatement';
  propertyValue: boolean;
  operator: BooleanPropertyOperator;
}

File: rulngin-api/src/main/java/com/techeule/filtereditor/api/rules/entity/property/BooleanPropertyStatement.java
package com.techeule.filtereditor.api.rules.entity.property;

public final class BooleanPropertyStatement extends PropertyStatement<Boolean> {

  @Override
  public PropertyOperator getOperator() {
    return PropertyOperator.PROPERTY_VALUE_EQUALS;
  }

  @Override
  public void setOperator(PropertyOperator operator) {
    super.setOperator(PropertyOperator.PROPERTY_VALUE_EQUALS);
  }

  @Override
  public boolean isValid() {
    return super.isValid() && this.propertyValues != null && this.propertyValues.size() == 1;
  }
}
